app/src/main/java/com/hemdan/mvvm/di/module/NetworkModule.kt

```diff
@@ -25,7 +33,18 @@ class NetworkModule {
     @Provides
     @Singleton
     fun providesOkHttpClient(httpLoggingInterceptor: HttpLoggingInterceptor): OkHttpClient {
-        return OkHttpClient.Builder().addInterceptor(httpLoggingInterceptor)
+        return OkHttpClient.Builder().addInterceptor { chain ->
+                val request = chain.request()
+                val url = request.url()
+                val newUrl= url.newBuilder().addQueryParameter(API_KEY, API_KEY_VALUE)
```

Comment:
@MonaAliMahmoud good think to use the interceptor but you can provide new interceptor like the HttpLoggingInterceptor provided